src/merge.cpp

```diff
@@ -0,0 +1,265 @@
+        [&](tbb::blocked_range<size_t> r)
+        {
+            for (size_t k = r.begin(); k < r.end(); ++k)
+            {
+                verify = chelper(Adfs[k], Bdfs[k]);
+
+                if (verify == false)
+                {
+                    return false;
+                }
+            }
+        },
+        ap);
+
+    return true;
+}
+
+MAT::Tree subtree(MAT::Tree tree, std::vector<std::string> common)
+{
+
+    tbb::mutex m;
+    auto tree_copy = get_tree_copy(tree);
+    auto all_leaves = tree_copy.get_leaves_ids();
+    std::vector<std::string> to_remove;
+    std::set_difference(all_leaves.begin(), all_leaves.end(), common.begin(), common.end(), std::back_inserter(to_remove));
+
+    for (auto k : to_remove)
+    {
+        tree_copy.remove_node(k, true);
+        // m.unlock();
+    }
+    return tree_copy;
+}
+bool chelper(MAT::Node *a, MAT::Node *b)
+{
+    tbb::mutex m;
+    concurMap::accessor ac;
+    if (a->is_root() && b->is_root())
+    {
+        return true;
+    }
+    if (a->mutations.size() != b->mutations.size())
+    {
+        return false;
+    }
+    for (int x = 0; x < a->mutations.size(); x++)
+    {
+        MAT::Mutation mut1 = a->mutations[x];
+        MAT::Mutation mut2 = b->mutations[x];
+        if (mut1.position != mut2.position)
+        {
+            return false;
+        }
+        else if (mut1.ref_nuc != mut2.ref_nuc)
+        {
+            return false;
+        }
+        else if (mut1.mut_nuc != mut2.mut_nuc)
+        {
+            return false;
+        }
+    }
+    // m.lock();
+    consistNodes.insert(ac, b->identifier);
+    ac->second = a->identifier;
+    ac.release();
+    //  m.unlock();
+    return true;
+}
+void merge_main(po::parsed_options parsed)
+{
+    timer.Start();
+    //Takes user input and loads the specified MAT files
+    po::variables_map vm = parse_merge_command(parsed);
+    std::string mat1_filename = vm["input-mat-1"].as<std::string>();
+    std::string mat2_filename = vm["input-mat-2"].as<std::string>();
+    std::string output_filename = vm["output-mat"].as<std::string>();
+    uint32_t num_threads = vm["threads"].as<uint32_t>();
+    tbb::task_scheduler_init init(num_threads);
+    MAT::Tree mat1 = MAT::load_mutation_annotated_tree(mat1_filename);
+    MAT::Tree mat2 = MAT::load_mutation_annotated_tree(mat2_filename);
+    MAT::Tree baseMat;
+    MAT::Tree otherMat;
+    concurMap::accessor ac;
+
+    if (mat1.condensed_nodes.size() > 0)
+    {
+        mat1.uncondense_leaves();
+    }
+    if (mat2.condensed_nodes.size() > 0)
+    {
+        mat2.uncondense_leaves();
+    }
+    //Assigns biggest MAT to baseMat and smaller one to otherMat
+    if (mat1.get_num_leaves() > mat2.get_num_leaves())
+    {
+        baseMat = mat1;
+        otherMat = mat2;
+    }
+    else
+    {
+        baseMat = mat2;
+        otherMat = mat1;
+    }
+    //Checks for consistency
+
+    if (consistent(baseMat, otherMat) == false)
+    {
```

Comment:
Your program should not return unexpectedly like this. Add a message why you quit (e.g. "ERROR: MATs not consistent") and use "exit(1)" instead of return since it is an error.